### PR TITLE
Prevent portal submissions on devnet or testnet

### LINKED ISSUE #136: `publish` subcommands should fail if the RPC URI looks like a devnet or testnet URI
Only MNB NFTs can be submitted to the dApp Publisher Portal. The `dapp-store` CLI should reject these submissions.

### PR DESCRIPTION
**Pull request by creativedrewy**
Fixes #136

### CODE REVIEW

File: packages/cli/src/utils.ts [hunk 1]
```diff
@@ -132,14 +132,32 @@ const checkImageExtension = (uri: string): boolean => {
...
 
+export const isDevnet = (rpcUrl: string): boolean => {
+  return rpcUrl.indexOf("devnet") != -1;
+};
+
+export const isTestnet = (rpcUrl: string): boolean => {
+  return rpcUrl.indexOf("testnet") != -1;
+};
+
+export const isMainnet = (rpcUrl: string): boolean => {
```
Comment (sdlaver) [inline packages/cli/src/utils.ts:143]: I don't know that this heuristic is accurate - I'd propose that mainnet is the default, in the absence of a signal that it is NOT mainnet

File: packages/cli/src/utils.ts [hunk 2]
```diff
@@ -132,14 +132,32 @@ const checkImageExtension = (uri: string): boolean => {
...
+  return rpcUrl.indexOf("mainnet") != -1;
+};
+
+export const checkSubmissionNetwork = (rpcUrl: string) => {
+  if (isDevnet(rpcUrl) || isTestnet(rpcUrl)) {
+    throw new Error("Thank you, but the Solana Mobile dApp publisher portal only supports mainnet submissions.");
```
Comment (sdlaver) [inline packages/cli/src/utils.ts:149]: I'd recommend we put a bit more detail into this message, and call out explicitly what caused the failure. Something like:

> It looks like you are attempting to publish an app with a devnet or testnet RPC endpoint. Please ensure that your app NFTs (publisher, app, and release) are minted on mainnet beta, and re-run this command with a mainnet beta RPC endpoint.

WDYT?

**Review by sdlaver — CHANGES_REQUESTED** [review]

File: packages/cli/src/utils.ts (hunk 1 above)
Comment (creativedrewy) [reply packages/cli/src/utils.ts:143]: Can do.

File: packages/cli/src/utils.ts (hunk 2 above)
Comment (creativedrewy) [reply packages/cli/src/utils.ts:149]: Do you think that is general to share the message across all of the portal-related commands - submit, update, etc?

**Review by sdlaver — APPROVED** [review]

**Pull request merged**